File: Python/1164-NumeroPerfeito.py
n = int(input())
for i in range(0,n):
    x = int(input())
    som = 0
    for j in range(1,x):
        if(j!=x):
            if(x%j==0):
                som+=j
    if(som==x):
        print(f'{x} eh perfeito')
    else:
        print(f'{x} nao eh perfeito')
    

File: Java/LinguagemC/1164 - Numero Perfeito.cpp
#include <iostream>
 
using namespace std;
 
int main() {
 
    int n, x;

    cin >> n;

    for(int i = 0; i < n; i++){
        cin >> x;
        int som=0;
        for(int j = 1; j <= x; j++){
            if(j!=x){
                if(x%j==0){
                    som+=j;
                }
            }           
        }
        if(som==x){
            cout << x << " eh perfeito" << endl;
        } else {
            cout << x << " nao eh perfeito" << endl;
        }
    }
 
    return 0;
}